### ISSUE TITLE
macOS ⌘H and ⌘M don’t work for Hide and Minimize (**regardless of webviews**)

### ISSUE THREAD

**Issue by babramczyk** [microsoft/vscode#133513]
# 🚨 Note

This issue was introduced around 2 months ago, and has been extremely disruptful and breaks OS-level behavior.

**❗ It is not a duplicate of https://github.com/microsoft/vscode/issues/71800 (which has been buried, as the original title suggested this only happened in webviews, which _is not the case_) ❗**

<hr />

Does this issue occur when all extensions are disabled?: Yes

VS Code Version: Version: 1.58.2
OS Version: macOS 11.5
Steps to Reproduce:

Randomly, many times on macOS the ⌘H and ⌘M commands don’t work for the Hide and Minimize actions.

This has been reported as others in this issue, which may not be getting enough attention, as it’s mis-labled at this point.

I have verified that it happens even when extensions are disabled, and that none of my keyboard shortcuts are conflicting here.

Sometimes happens, sometimes doesn’t. If I quit VS Code and re-open it, it typically fixes it

Edit: Some extra context: currently (and I think this is the case frequently), this is only a problem for 1 window. i.e. all VS Code windows can Hide and Minimize properly, except for one. (it could be the first one I opened..? But that’s just a guess)

**Comment by lramos15** [comment]
Webviews occur in a lot of places are you sure you didn't open a markdown preview or anything like that? I cannot reproduce this issue without a webview.

**Comment by babramczyk** [comment]
@lramos15 yes. I have consistently witnessed this behavior throughout the past few months, many times confirming that I have not opened any webviews, nor do I have any webviews open at the time of trying to minimize

FWIW, I typically have many windows of VS Code open (but typically most are minimized). Maybe that is part of the problem..?

**Comment by scztt** [comment]
I'm seeing this issue as well. It appears to be affected by some non-obvious state of VSCode - I see this behavior:

1. At some point after working for a while, I observe that Cmd+H / Cmd+M shortcuts no longer work.
2. If I open a new window (File -> New Window), these shortcuts work when the new window is focused, but not the old.
3. They continue to work in the new window even when I match the state to the old (non-working) window (e.g. focus on the Search panel, show the Output panel etc).
4. If I close my old non-working window and re-open it, Cmd+H / Cmd+M begin to work again.

**Comment by sbatten** [comment]
When you can repro, can you verify if the menu entries for minimize and hide appear in the application menu? Also, do those menu entries work (by clicking them instead of using the keybinding)?

**Comment by babramczyk** [comment]
@sbatten yes, they do still appear and function (I’ve been using the menu entries instead of the keybindings for a few months now -- also just double checked to make sure)

**Comment by jon-sully** [comment]
Can confirm, it's happening quite a bit to me as well — the menu options for hide and minimize work fine but hitting the key commands do not

**Comment by matkovsky** [comment]
https://github.com/microsoft/vscode/issues/135169#issuecomment-977718324

**Comment by bgreenlee** [comment]
I just started seeing this too. It works from the menu, but the shortcut is greyed out and doesn't work.

![image](https://user-images.githubusercontent.com/2901/145459376-913a3950-67d9-416f-847a-7a33106bc436.png)

Version: 1.64.0-insider (Universal)
Commit: 68976705968229accd11fdfe1cba36a908208dc9
Date: 2021-12-09T05:14:38.594Z
Electron: 13.5.2
Chromium: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Darwin x64 21.1.0

**Comment by jcollum** [comment]
Confirmed here #139649 -- have some keybinding listener output there as well. This happens when I am editing a text file.

**Comment by oscarrutt** [comment]
I started #142900 which was closed and referred here as a duplicate.

On #142900 @alexdima asked me, "Please run `Toggle Keyboard Shortcuts Troubleshooting` when this happens next and include the output here" so I am responding here now.

While the following troubleshooting output was generated, I pressed, in this order, Command-H, Command-M, and Option-Command-H.

[2022-02-23 14:45:11.128] [renderer2] [info] [KeybindingService]: / Received  keydown event - modifiers: [meta], code: MetaLeft, keyCode: 91, key: Meta
[2022-02-23 14:45:11.129] [renderer2] [info] [KeybindingService]: | Converted keydown event - modifiers: [meta], code: MetaLeft, keyCode: 57 ('Meta')
[2022-02-23 14:45:11.129] [renderer2] [info] [KeybindingService]: \ Keyboard event cannot be dispatched in keydown phase.
[2022-02-23 14:45:11.489] [renderer2] [info] [KeybindingService]: / Received  keydown event - modifiers: [meta], code: KeyH, keyCode: 72, key: h
[2022-02-23 14:45:11.489] [renderer2] [info] [KeybindingService]: | Converted keydown event - modifiers: [meta], code: KeyH, keyCode: 38 ('H')
[2022-02-23 14:45:11.490] [renderer2] [info] [KeybindingService]: | Resolving meta+[KeyH]
[2022-02-23 14:45:11.490] [renderer2] [info] [KeybindingService]: \ From 1 keybinding entries, no when clauses matched the context.
[2022-02-23 14:45:11.704] [renderer2] [info] [KeybindingService]: + Storing single modifier for possible chord meta.
[2022-02-23 14:45:12.010] [renderer2] [info] [KeybindingService]: + Clearing single modifier due to 300ms elapsed.
[2022-02-23 14:45:13.360] [renderer2] [info] [KeybindingService]: / Received  keydown event - modifiers: [meta], code: MetaLeft, keyCode: 91, key: Meta
[2022-02-23 14:45:13.361] [renderer2] [info] [KeybindingService]: | Converted keydown event - modifiers: [meta], code: MetaLeft, keyCode: 57 ('Meta')
[2022-02-23 14:45:13.361] [renderer2] [info] [KeybindingService]: \ Keyboard event cannot be dispatched in keydown phase.
[2022-02-23 14:45:14.072] [renderer2] [info] [KeybindingService]: / Received  keydown event - modifiers: [meta], code: KeyM, keyCode: 77, key: m
[2022-02-23 14:45:14.072] [renderer2] [info] [KeybindingService]: | Converted keydown event - modifiers: [meta], code: KeyM, keyCode: 43 ('M')
[2022-02-23 14:45:14.073] [renderer2] [info] [KeybindingService]: | Resolving meta+[KeyM]
[2022-02-23 14:45:14.073] [renderer2] [info] [KeybindingService]: \ From 8 keybinding entries, no when clauses matched the context.
[2022-02-23 14:45:14.360] [renderer2] [info] [KeybindingService]: + Storing single modifier for possible chord meta.
[2022-02-23 14:45:14.665] [renderer2] [info] [KeybindingService]: + Clearing single modifier due to 300ms elapsed.
[2022-02-23 14:45:15.593] [renderer2] [info] [KeybindingService]: / Received  keydown event - modifiers: [meta], code: MetaLeft, keyCode: 91, key: Meta
[2022-02-23 14:45:15.594] [renderer2] [info] [KeybindingService]: | Converted keydown event - modifiers: [meta], code: MetaLeft, keyCode: 57 ('Meta')
[2022-02-23 14:45:15.594] [renderer2] [info] [KeybindingService]: \ Keyboard event cannot be dispatched in keydown phase.
[2022-02-23 14:45:15.639] [renderer2] [info] [KeybindingService]: / Received  keydown event - modifiers: [alt,meta], code: AltLeft, keyCode: 18, key: Alt
[2022-02-23 14:45:15.640] [renderer2] [info] [KeybindingService]: | Converted keydown event - modifiers: [alt,meta], code: AltLeft, keyCode: 6 ('Alt')
[2022-02-23 14:45:15.641] [renderer2] [info] [KeybindingService]: \ Keyboard event cannot be dispatched in keydown phase.
[2022-02-23 14:45:15.982] [renderer2] [info] [KeybindingService]: / Received  keydown event - modifiers: [alt,meta], code: KeyH, keyCode: 72, key: ˙
[2022-02-23 14:45:15.988] [renderer2] [info] [KeybindingService]: | Converted keydown event - modifiers: [alt,meta], code: KeyH, keyCode: 38 ('H')
[2022-02-23 14:45:15.990] [renderer2] [info] [KeybindingService]: | Resolving alt+meta+[KeyH]
[2022-02-23 14:45:15.993] [renderer2] [info] [KeybindingService]: \ No keybinding entries.
[2022-02-23 14:45:16.208] [renderer2] [info] [KeybindingService]: + Ignoring single modifier meta due to it being pressed together with other keys.
[2022-02-23 14:45:24.400] [renderer2] [info] [KeybindingService]: / Received  keydown event - modifiers: [meta], code: MetaLeft, keyCode: 91, key: Meta
[2022-02-23 14:45:24.466] [renderer2] [info] [KeybindingService]: | Converted keydown event - modifiers: [meta], code: MetaLeft, keyCode: 57 ('Meta')
[2022-02-23 14:45:24.466] [renderer2] [info] [KeybindingService]: \ Keyboard event cannot be dispatched in keydown phase.
[2022-02-23 14:45:24.466] [renderer2] [info] [KeybindingService]: / Received  keydown event - modifiers: [shift,meta], code: ShiftLeft, keyCode: 16, key: Shift
[2022-02-23 14:45:24.467] [renderer2] [info] [KeybindingService]: | Converted keydown event - modifiers: [shift,meta], code: ShiftLeft, keyCode: 4 ('Shift')
[2022-02-23 14:45:24.468] [renderer2] [info] [KeybindingService]: \ Keyboard event cannot be dispatched in keydown phase.
[2022-02-23 14:45:24.752] [renderer2] [info] [KeybindingService]: / Received  keydown event - modifiers: [shift,meta], code: KeyP, keyCode: 80, key: p
[2022-02-23 14:45:24.752] [renderer2] [info] [KeybindingService]: | Converted keydown event - modifiers: [shift,meta], code: KeyP, keyCode: 46 ('P')
[2022-02-23 14:45:24.753] [renderer2] [info] [KeybindingService]: | Resolving shift+meta+[KeyP]
[2022-02-23 14:45:24.753] [renderer2] [info] [KeybindingService]: \ From 2 keybinding entries, matched workbench.action.showCommands, when: no when condition, source: built-in.
[2022-02-23 14:45:24.975] [renderer2] [info] [KeybindingService]: + Ignoring single modifier meta due to it being pressed together with other keys.

**Comment by marcleblanc2** [comment]
I see the same. It's almost like VSCode is misinterpreting the cmd modifier for meta. It's the same for the MacBook's keyboard, and my external keyboard. This is the keyboard shortcut troubleshooting log output for one cmd+h:

[2022-04-13 23:09:51.990] [renderer1] [info] [KeybindingService]: / Received  keydown event - modifiers: [meta], code: MetaLeft, keyCode: 91, key: Meta
[2022-04-13 23:09:51.991] [renderer1] [info] [KeybindingService]: | Converted keydown event - modifiers: [meta], code: MetaLeft, keyCode: 57 ('Meta')
[2022-04-13 23:09:51.991] [renderer1] [info] [KeybindingService]: \ Keyboard event cannot be dispatched in keydown phase.
[2022-04-13 23:09:52.278] [renderer1] [info] [KeybindingService]: / Received  keydown event - modifiers: [meta], code: KeyH, keyCode: 72, key: h
[2022-04-13 23:09:52.278] [renderer1] [info] [KeybindingService]: | Converted keydown event - modifiers: [meta], code: KeyH, keyCode: 38 ('H')
[2022-04-13 23:09:52.279] [renderer1] [info] [KeybindingService]: | Resolving meta+[KeyH]
[2022-04-13 23:09:52.279] [renderer1] [info] [KeybindingService]: \ No keybinding entries.
[2022-04-13 23:09:53.173] [renderer1] [info] [KeybindingService]: + Storing single modifier for possible chord meta.
[2022-04-13 23:09:53.476] [renderer1] [info] [KeybindingService]: + Clearing single modifier due to 300ms elapsed.

**Comment by alexdima** [comment]
> It's almost like VSCode is misinterpreting the cmd modifier for meta. It's the same for the MacBook's keyboard, and my external keyboard. This is the keyboard shortcut troubleshooting log output for one cmd+h

`cmd+h` is not dispatched by vscode, it is dispatched by the OS or Electron, so `cmd+H` will also appear as having no keybindings entries even on machines where `cmd+H` works to hide the window.

**Comment by BB9z** [comment]
I have a workaround:

Open command palette and search: Open Keyboard Shortcuts (JSON)

Then put this item in config file:

```json
{
    "key": "",
    "command": "testing.toggleTestingPeekHistory"
}
```

Restart VSCode then it will be fine.

**Comment by bradchoate** [comment]
Ah, thanks @BB9z — that helped me locate an existing default bind of Cmd+H to `Test: Toggle Test History in Peek`. I removed that binding, and haven't had an issue since.

**Comment by oscarrutt** [comment]
@bradchoate  Does that also fix the other problems such as Command-M?

**Comment by bradchoate** [comment]
> @bradchoate Does that also fix the other problems such as Command-M?

I'm rarely use that key, but no apparent problems with it (but I also am not seeing any mappings for Command+M, either).

**Comment by nstanger** [comment]
> Ah, thanks @BB9z — that helped me locate an existing default bind of Cmd+H to `Test: Toggle Test History in Peek`. I removed that binding, and haven't had an issue since.

But the default for that binding is **Option**-H, not Cmd-H. If that’s blocking Cmd-H then this is a much bigger problem…

FWIW simply disabling the `testing.toggleTestingPeekHistory` keybinding has no effect without a restart, so it seems to me more likely that it’s the restart that fixes things.

(And add me to the chorus of people with this issue — I mostly run in full screen so I hadn’t noticed it until a couple of days ago, and now it’s driving me bonkers.)

**Comment by oscarrutt** [comment]
> FWIW simply disabling the testing.toggleTestingPeekHistory keybinding has no effect without a restart, so it seems to me more likely that it’s the restart that fixes things.

Yes, the behavior of restarting _temporarily_ fixing the problem has been noted, either in this thread or in one of the  threads which fed into this one.

Also, it has been previously noted that the problem seems to affect only a main workspace window. A Duplicate Workspace window is not affected, for example,  IIRC. Please, someone confirm this so I don't start any false symptoms.

**Comment by deepak1556** [comment]
This should fixed in latest stable `1.73` with Electron 19.
